microapps/FeatureFormsApp/app/src/main/java/com/arcgismaps/toolkit/featureformsapp/screens/map/MapViewModel.kt

```diff
@@ -155,6 +163,23 @@ class MapViewModel @Inject constructor(
         )
         return Result.success(Unit)
     }
+    
+    fun selectNewFeature() {
+        require(_uiState.value is UIState.Switching)
```

Comment:
```suggestion
        if(_uiState.value is UIState.Switching)
```
It might be better to use an `if` here since this can be invoked from UI.